File Drag and Drop support

jkelleyrtp opened the pull request:
This PR exposes drag and drop support for files in Dioxus Desktop.

cc #83

higumachan commented:
I incorporated this PR into my repository and got a screen like this

![image](https://user-images.githubusercontent.com/1011298/148637776-33abfcce-e23c-4568-871a-baac648950be.png)

--- 

My environment

OS: Mac Big Sur 11.2.3
rustc: rustc 1.57.0 (f1edd0429 2021-11-29)
Cargo setting: dioxus = { git = "https://github.com/DioxusLabs/dioxus.git", branch = "jk/filedragindrop", features = ["desktop"] }

--- 
mycode

```rust
use dioxus::prelude::*;

pub fn start_app() {
    dioxus::desktop::launch(app);
}

fn app(cx: Scope) -> Element {
    let url = include_str!("../assets/lenna.url");

    cx.render(rsx!{
        div {
            //img {
            //    src: "{url}",
            //}
            input {
                r#type: "file",
                onchange: move |formdata| { println!("{:?}", formdata.data) },
            }
        }
    })
}
```

jkelleyrtp commented:
> I incorporated this PR into my repository and got a screen like this
> 
> ![image](https://user-images.githubusercontent.com/1011298/148637776-33abfcce-e23c-4568-871a-baac648950be.png)
> 
> My environment
> 
> OS: Mac Big Sur 11.2.3 rustc: rustc 1.57.0 (f1edd0429 2021-11-29) Cargo setting: dioxus = { git = "https://github.com/DioxusLabs/dioxus.git", branch = "jk/filedragindrop", features = ["desktop"] }
> 
> mycode
> 
> ```rust
> use dioxus::prelude::*;
> 
> pub fn start_app() {
>     dioxus::desktop::launch(app);
> }
> 
> fn app(cx: Scope) -> Element {
>     let url = include_str!("../assets/lenna.url");
> 
>     cx.render(rsx!{
>         div {
>             //img {
>             //    src: "{url}",
>             //}
>             input {
>                 r#type: "file",
>                 onchange: move |formdata| { println!("{:?}", formdata.data) },
>             }
>         }
>     })
> }
> ```

Ah yep, you're right. We were navigating the webview to the wrong internal URL. Thank you for testing it. I've updated the PR to fix it - it works on my machine.

higumachan commented:
It's working!
